pdns/dnsdistdist/dnsdist-self-answers.cc

```diff
@@ -240,4 +240,32 @@ bool generateAnswerFromRawPacket(DNSQuestion& dnsQuestion, const PacketBuffer& p
   return true;
 }
 
+bool removeRecordsAndSetRCode(DNSQuestion& dnsQuestion, uint8_t rcode)
+{
+  bool dnssecOK = false;
+  bool hadEDNS = false;
+  if (dnsdist::configuration::getCurrentRuntimeConfiguration().d_addEDNSToSelfGeneratedResponses && queryHasEDNS(dnsQuestion)) {
+    hadEDNS = true;
+    dnssecOK = ((dnsdist::getEDNSZ(dnsQuestion) & EDNS_HEADER_FLAG_DO) != 0);
+  }
+
+  dnsdist::PacketMangling::editDNSHeaderFromPacket(dnsQuestion.getMutableData(), [rcode](dnsheader& header) {
+    header.rcode = rcode;
+    header.qr = true;
+    header.qdcount = htons(1);
+    header.arcount = 0;
+    header.nscount = 0;
+    header.ancount = 0;
```

Comment:
Just so they are in wire-format order:
```suggestion
    header.ancount = 0;
    header.nscount = 0;
    header.arcount = 0;
```